File: app/src/main/java/com/stackanswer/source/local/room/moviefavorite/MovieFavoriteDao.java
package com.stackanswer.source.local.room.moviefavorite;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface MovieFavoriteDao {

    @Query("SELECT * FROM movie_table ORDER BY id ASC")
    DataSource.Factory<Integer, MovieFavorite> getAllMovie();

    @Insert(onConflict= OnConflictStrategy.REPLACE)
    void insert(MovieFavorite movieFavorite);

    @Update
    void update(MovieFavorite movieFavorite);

    @Delete
    void delete(MovieFavorite movieFavorite);

    @Query("DELETE FROM movie_table")
    void deleteAllMovie();
}

File: app/src/main/java/com/stackanswer/source/local/room/moviefavorite/MovieFavoriteDao.kt
package com.stackanswer.source.local.room.moviefavorite

import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Flowable

@Dao
interface MovieFavoriteDao {

    @Query("SELECT * FROM movie_table ORDER BY id ASC")
    fun getAllMovie(): Flowable<List<MovieFavorite>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(movieFavorite: MovieFavorite): Completable

    @Update
    fun update(movieFavorite: MovieFavorite?)

    @Delete
    fun delete(movieFavorite: MovieFavorite?)

    @Query("DELETE FROM movie_table")
    fun deleteAllMovie()
}